naga/tests/out/glsl/spec-constants-issue-5598.fragment.Fragment.glsl

```diff
@@ -0,0 +1,34 @@
+#version 310 es
+
+precision highp float;
+precision highp int;
+
+vec4 global = vec4(0.0);
+
+vec4 global_1 = vec4(0.0);
+
+layout(location = 0) out vec4 _fs2p_location0;
+
+void function() {
+    vec2 phi_52_ = vec2(0.0);
+    vec4 _e7 = global;
+    if (false) {
+        phi_52_ = vec2((_e7.x * 0.5), (_e7.y * 0.5));
```

Comment:
Is eliminating the dead branch in scope for naga? Or that's a job for the target backend compiler?

Reply:
Naga isn't generally in the optimization business. That would be a huge expansion of scope. Most drivers are running LLVM or the equivalent anyway.